readthedocs/core/resolver.py

```diff
@@ -144,13 +144,24 @@ def resolve(self, project, protocol='http', filename='', private=None,
                 version_slug = project.get_default_version()
             private = self._get_private(project, version_slug)
 
-        domain = self.resolve_domain(project, private=private)
+        canonical_project = self._get_canonical_project(project)
+        custom_domain = self._get_project_custom_domain(canonical_project)
 
-        # Use HTTPS if settings specify
-        public_domain = getattr(settings, 'PUBLIC_DOMAIN', None)
-        use_https = getattr(settings, 'PUBLIC_DOMAIN_USES_HTTPS', False)
-        if use_https and public_domain and public_domain in domain:
-            protocol = 'https'
+        if custom_domain:
+            domain = custom_domain.domain
+        elif self._use_subdomain():
+            domain = self._get_project_subdomain(canonical_project)
+        else:
+            domain = getattr(settings, 'PRODUCTION_DOMAIN')
```

Comment:
Isn't this `if/elif/else` the same code of what `self.resolve_domain` does?

Reply:
It is. This function needs access to the `Domain` object in order to read the `.https` property. That object is not exposed from the `resolve_domain` method.

Reply:
Loading that object again based on the hostname alone will require another database query and the resolver is already the largest single source of database load in Read the Docs.

Reply:
I understand and I saw that you mentioned the issue to refactor this. So, I'm fine with "duplication" here.

Although, I think that adding a similar explanation written in these comments as a comment in the code would make it easier to refactor this in the future.